src/D2L.CodeStyle.Analyzers/ApiUsage/RpcAnalyzer.cs

```diff
@@ -102,6 +129,54 @@ INamedTypeSymbol dependencyAttributeType
 				// other things to check:
 				// - appropriate use of [Dependency]
 			}
+
+			CheckThatRpcParametersAreValid(
+				context: context,
+				ps: method.ParameterList.Parameters,
+				dependencyAttributeType: dependencyAttributeType,
+				deserializableType: deserializableType,
+				deserializerType: deserializerType,
+				dictionaryType: dictionaryType,
+				knownRpcParameterTypes: knownRpcParameterTypes
+			);
+		}
+
+		private static void CheckThatRpcParametersAreValid(
+			SyntaxNodeAnalysisContext context,
+			SeparatedSyntaxList<ParameterSyntax> ps,
+			INamedTypeSymbol dependencyAttributeType,
+			INamedTypeSymbol deserializableType,
+			INamedTypeSymbol deserializerType,
+			INamedTypeSymbol dictionaryType,
+			ImmutableHashSet<INamedTypeSymbol> knownRpcParameterTypes
+		) {
+			foreach( var parameter in ps.Skip( 1 ) ) {
+				// Skip injected parameters
+				if( dependencyAttributeType != null ) {
+					if( IsMarkedDependency(
+						dependencyAttributeType: dependencyAttributeType,
+						parameter: parameter,
+						model: context.SemanticModel
+					) ) {
+						continue;
+					}
+				}

```

Comment:
Could do the same "splitter" approach activation diagnostics was doing (separating parameters into a tuple of (context, dependencies, parameters), instead of duplicating this enumeration and check.

Reply:
Ah yeah, same thing I was mentioning up above. You could split or just do a big loop that checks both things.